### DISCUSSION TITLE
Authorisation

### DISCUSSION THREAD

**Discussion by VagueGit** [Mimetis/Dotmim.Sync#379] [General]
Two issues which address auth/security for the DMS sync rest api: Supports authentication #42 and WebServer Authorization #326 

Auth is outside the scope of DMS but auth is a big deal for everyone. It would be nice if the docs included an example of auth, perhaps bringing #42 and #326 together with current examples.

AzureAD has changed a bit since and there is a lot a user can get wrong if they only want to know enough for DMS.

I think it would help jump start users of DMS if we didn't have to figure out auth for ourselves.

In the interim this discussion may be helpful to users getting started with DMS who wish to implement auth.

**EDIT 2** The example given in #42 `authenticationResult = await publicClientApp.AcquireTokenAsync(scopes);` has been deprecated
`Error CS0619 'IPublicClientApplication.AcquireTokenAsync(IEnumerable<string>)' is obsolete: 'Use AcquireTokenInteractive instead. See https://aka.ms/msal-net-3-breaking-changes. '`

Of course we don't want the user to interact here. Guidance would be much appreciated.

**Comment by Mimetis** [comment]
https://github.com/Mimetis/Dotmim.Sync/discussions/393
